### PR TITLE
Laboratory work #2, Aisina Alina - 21FPL1

### PR DESCRIPTION
**Pull request by itsaisina**

### CODE REVIEW

File: lab_2_keywords_cooccurrence/main.py [hunk 1]
```diff
@@ -55,7 +120,16 @@ def calculate_word_degrees(candidate_keyword_phrases: KeyPhrases,
...
     """
-    pass
+    if not correct_list(candidate_keyword_phrases, tuple, False) or not correct_list(content_words, str, False):
+        return None
+    word_degrees = {}
+    for phrase in candidate_keyword_phrases:
+        for word in content_words:
+            if word in phrase:
+                word_degrees[word] = word_degrees.get(word, 0) + len(phrase)
+            elif word not in word_degrees.keys():
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:130]:
```suggestion
            elif word not in word_degrees:
```

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/main.py (hunk 1 above)
Comment (itsaisina) [reply lab_2_keywords_cooccurrence/main.py]: done

File: lab_2_keywords_cooccurrence/main.py [hunk 2]
```diff
@@ -141,7 +266,21 @@ def calculate_cumulative_score_for_candidates_with_stop_words(candidate_keyword_
...
-    pass
+    if not correct_list(candidate_keyword_phrases, tuple, False) or not correct_dict(word_scores, str, float, False) \
+            or not correct_list(stop_words, str, False):
+        return None
+    result_dict = {}
+    for item in candidate_keyword_phrases:
+        count = 0
+        for elem in item:
+            for key, value in word_scores.items():
+                if elem == key:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:277]: reverse the condition and continue

File: lab_2_keywords_cooccurrence/main.py [hunk 3]
```diff
@@ -123,7 +213,42 @@ def extract_candidate_keyword_phrases_with_adjoining(candidate_keyword_phrases:
...
+                count += 1
+        if count >= 2:
+            final_list.append(item)
+    result = list(set(final_list))[::-1]
+
+    key_phrases = []
+    for item1 in phrases:
+        for elem in result:
+            first_tuple, second_tuple = elem[0][0], elem[1][len(elem[1]) - 1]
+            if first_tuple in item1:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:238]: reverse condition and continue

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/main.py (hunk 3 above)
Comment (itsaisina) [reply lab_2_keywords_cooccurrence/main.py]: Done

File: lab_2_keywords_cooccurrence/main.py [hunk 4]
```diff
@@ -3,12 +3,41 @@
...
 """
 from pathlib import Path
-from typing import Optional, Sequence, Mapping
+from typing import Optional, Sequence, Mapping, Any
 
 KeyPhrase = tuple[str, ...]
 KeyPhrases = Sequence[KeyPhrase]
 
 
+def correct_list(list1: Any, type1: Any, empty: bool) -> bool:
```
[the remark this answers is not in the document]
Comment (itsaisina) [reply lab_2_keywords_cooccurrence/main.py]: done

File: lab_1_keywords_tfidf/main.py [hunk 5]
```diff
@@ -21,7 +21,7 @@ def check_list(user_input: Any, elements_type: type, can_be_empty: bool) -> bool
     return True
 
 
-def check_dict(user_input: dict, key_type: type, value_type: type, can_be_empty: bool) -> bool:
+def check_dict(user_input: Any, key_type: type, value_type: Any, can_be_empty: bool) -> bool:
     """
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:25]: too many changes

File: lab_2_keywords_cooccurrence/main.py [hunk 6]
```diff
@@ -17,7 +19,13 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
...
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(text, str) or not text:
+        return None
+    punctuation_marks = '''.,;':¡!¿?…⋯‹›«»\\/"“”[]()⟨⟩}{&|-–~—'''
+    for mark in punctuation_marks:
+        text = text.replace(mark, ',')
+    split_text = text.split(',')
+    return [phrase.strip() for phrase in split_text if phrase.strip() != '']
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:28]: `!= ''` not needed.  and use := operation

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_keywords_tfidf/main.py (hunk 5 above)
Comment (itsaisina) [reply lab_1_keywords_tfidf/main.py:25]: done

**Pull request closed without merging**